presto-native-execution/presto_cpp/main/PrestoTask.cpp

```diff
@@ -265,7 +265,25 @@ protocol::TaskInfo PrestoTask::updateInfoLocked() {
 
   // TODO(venkatra): Populate these memory stats as well.
   prestoTaskStats.revocableMemoryReservationInBytes = {};
-  prestoTaskStats.cumulativeUserMemory = {};
+
+  // Set the lastTaskStatsUpdateMs to execution start time if it is 0.
+  if (lastTaskStatsUpdateMs == 0) {
+    lastTaskStatsUpdateMs = taskStats.executionStartTimeMs;
+  }
+
+  const uint64_t currentTimeMs = velox::getCurrentTimeMs();
+  const uint64_t lastPeriodMs = currentTimeMs - lastTaskStatsUpdateMs;
+
+  const int64_t currentBytes = tracker->currentBytes();
+
+  int64_t averageMemoryForLastPeriod =
+      (currentBytes + lastMemoryReservation) / 2;
+
+  prestoTaskStats.cumulativeUserMemory +=
```

Comment:
Can we also update the prestoTaskStats.cumulativeTotalMemory to be the same as prestoTaskStats.cumulativeUserMemory?